src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -257,133 +263,199 @@ private void enqueueNotification(TripMonitorNotification ...tripMonitorNotificat
+        long millisSinceLastCheck = DateTimeUtils.currentTimeMillis() - journeyState.lastCheckedMillis;
         long minutesSinceLastCheck = TimeUnit.MILLISECONDS.toMinutes(millisSinceLastCheck);
-        LOG.info("{} minutes since last check", minutesSinceLastCheck);
-        long minutesUntilTrip = Duration.between(now, tripTime).toMinutes();
-        LOG.info("Trip {} starts in {} minutes", trip.id, minutesUntilTrip);
-        // TODO: Refine these frequency intervals for monitor checks.
+        LOG.info("{} minutes since last checking trip", minutesSinceLastCheck);
+        long minutesUntilTrip = (tripStartInstant.getEpochSecond() - now.toEpochSecond()) / 60;
+        LOG.info("Trip starts in {} minutes", minutesUntilTrip);
+        // skip check if the time until the next trip starts is longer than the requested lead time
+        if (minutesUntilTrip > trip.leadTimeInMinutes) {
+            LOG.info(
+                "The time until this trip begins again is more than the {}-minute lead time. Skipping trip.",
+                trip.leadTimeInMinutes
+            );
+            return true;
+        }
         // If time until trip is greater than 60 minutes, we only need to check once every hour.
         if (minutesUntilTrip > 60) {
             // It's been about an hour since the last check. Do not skip.
             if (minutesSinceLastCheck >= 60) {
                 // TODO: Change log level.
-                LOG.info("Trip {} not checked in at least an hour. Checking.", trip.id);
+                LOG.info("Trip not checked in at least an hour. Checking.");
                 return false;
             }
         } else {
             // It's less than an hour until the trip time, start more frequent trip checks (about every 15 minutes).
             if (minutesSinceLastCheck >= 15) {
                 // Last check was more than 15 minutes ago. Check. (approx. 4 checks per hour).
                 // TODO: Change log level.
-                LOG.info("Trip {} happening soon. Checking.", trip.id);
+                LOG.info("Trip happening soon. Checking.");
                 return false;
             }
-            // If the minutes until the trip is within the lead time, check the trip every minute
-            // (assuming the loop runs every minute).
-            // TODO: Should lead time just be used for notification time limit?
-            if (minutesUntilTrip <= trip.leadTimeInMinutes) {
+            // If the trip starts within 30 minutes, check the trip every minute (assuming the loop runs every minute).
+            if (minutesUntilTrip <= 30) {
                 // TODO: Change log level.
-                LOG.info(
-                    "Trip {} happening within user's lead time ({} minutes). Checking every minute.",
-                    trip.id,
-                    trip.leadTimeInMinutes
-                );
+                LOG.info("Trip happening within 30 minutes. Checking every minute.");
```

Comment:
This should use a variable that is also used in the conditional check (so that they don't become out of sync).

Reply:
Added in 6fd00a976c063c5b46e7254a995fbbb628b90445